src/main/java/io/github/enkarin/bookcrossing/chat/service/CorrespondenceService.java

```diff
@@ -64,6 +63,7 @@ public void deleteChat(final int userId, final String login) {
         correspondenceRepository.delete(correspondence);
     }
 
+    @Transactional
```

Comment:
Why?
Looks like an error

Reply:
Consider to move out `getMessages` to separate class instead of starting write transaction here